src/control/tools/EditSelectionContents.cpp

```diff
@@ -380,21 +381,31 @@ void EditSelectionContents::finalizeSelection(double x, double y, double width,
 
 	bool move = mx != 0 || my != 0;
 
-	for (Element* e: this->selected)
+
+	for (auto e = this->selected.rbegin(); e != this->selected.rend(); ++e)
 	{
 		if (move)
 		{
-			e->move(mx, my);
+			(*e)->move(mx, my);
```

Comment:
The repeated `*e` may seem confusing/error prone to a reader at first glance. I suggest using `auto it` in the for loop and explicitly setting `Element* e = *it` so it's clear that `e` is, in fact, an `Element*`.